trinity/utils/chains.py

```diff
@@ -60,55 +62,61 @@ def get_data_dir_for_network_id(network_id: int) -> str:
 NODEKEY_FILENAME = 'nodekey'
 
 
-def get_nodekey_path(data_dir: str) -> str:
+def get_nodekey_path(data_dir: PurePath) -> PurePath:
     """
     Returns the path to the private key used for devp2p connections.
     """
-    return os.environ.get(
+    return PurePath(os.environ.get(
         'TRINITY_NODEKEY',
-        os.path.join(data_dir, NODEKEY_FILENAME),
-    )
+        str(data_dir / NODEKEY_FILENAME),
+    ))
 
 
 DATABASE_SOCKET_FILENAME = 'db.ipc'
 
 
-def get_database_socket_path(data_dir: str) -> str:
+def get_database_socket_path(data_dir: PurePath) -> str:
     """
     Returns the path to the private key used for devp2p connections.
+
+    We're still returning 'str' here on ipc-related path because an issue with
+    multi-processing not being able to interpret 'PurePath' objects correctly.
     """
     return os.environ.get(
         'TRINITY_DATABASE_IPC',
-        os.path.join(data_dir, DATABASE_SOCKET_FILENAME),
+        os.path.join(str(data_dir), DATABASE_SOCKET_FILENAME),
     )
 
 
 JSONRPC_SOCKET_FILENAME = 'jsonrpc.ipc'
 
 
-def get_jsonrpc_socket_path(data_dir: str) -> str:
+def get_jsonrpc_socket_path(data_dir: PurePath) -> str:
     """
     Returns the path to the ipc socket for the JSON-RPC server.
+
+    We're still returning 'str' here on ipc-related path because an issue with
+    multi-processing not being able to interpret 'PurePath' objects correctly.
     """
     return os.environ.get(
         'TRINITY_JSONRPC_IPC',
-        os.path.join(data_dir, JSONRPC_SOCKET_FILENAME),
+        os.path.join(str(data_dir), JSONRPC_SOCKET_FILENAME),
     )
 
 
 #
 # Nodekey loading
 #
-def load_nodekey(nodekey_path: str) -> PrivateKey:
-    with open(nodekey_path, 'rb') as nodekey_file:
+def load_nodekey(nodekey_path: PurePath) -> PrivateKey:
+    with open(str(nodekey_path), 'rb') as nodekey_file:
```

Comment:
```py
with nodekey_path.open('rb') as nodekey_file:
```
[according to docs](https://docs.python.org/3/library/pathlib.html#pathlib.Path.open)